diff --git a/build.sh b/build.sh
index b57851b..c97ff4f 100755
--- a/build.sh
+++ b/build.sh
@@ -166,16 +166,16 @@ if buildAll || hasArg libengine; then
 
     echo "Building libengine: make step"
     if [[ ${TESTS} == "ON" ]]; then
-        echo "make -j4 all"
-        make -j4 all
+        echo "make -j${PARALLEL_LEVEL} all"
+        make -j${PARALLEL_LEVEL} all
     else
-        echo "make -j4 blazingsql-engine VERBOSE=${VERBOSE}"
-        make -j4 blazingsql-engine VERBOSE=${VERBOSE}
+        echo "make -j${PARALLEL_LEVEL} blazingsql-engine VERBOSE=${VERBOSE}"
+        make -j${PARALLEL_LEVEL} blazingsql-engine VERBOSE=${VERBOSE}
     fi
 
     if [[ ${INSTALL_TARGET} != "" ]]; then
-        echo "make -j4 install VERBOSE=${VERBOSE}"
-        make -j4 install VERBOSE=${VERBOSE}
+        echo "make -j${PARALLEL_LEVEL} install VERBOSE=${VERBOSE}"
+        make -j${PARALLEL_LEVEL} install VERBOSE=${VERBOSE}
         cp libblazingsql-engine.so ${INSTALL_PREFIX}/lib/libblazingsql-engine.so
     fi
 fi
