diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index f1bfc19..74ee2b3 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -118,11 +118,12 @@ else()
   message(FATAL_ERROR "ZLib not found, please check your settings.")
 endif(ZLIB_FOUND)

-if(NOT DEFINED BLAS_LIBRARIES)
-  find_package( BLAS REQUIRED )
+find_package(BLAS REQUIRED)
+if(BLAS_FOUND)
+  message(STATUS "BLAS found in ${BLAS_LIBRARIES}")
 else()
-  message(STATUS "Manually setting BLAS to ${BLAS_LIBRARIES}")
-endif()
+  message(FATAL_ERROR "BLAS not found, please check your settings.")
+endif(BLAS_FOUND)

 set(Protobuf_USE_STATIC_LIBS ON)
 find_package(Protobuf REQUIRED)
@@ -411,6 +412,7 @@ if(BUILD_CUML_CPP_LIBRARY)
     treelitelib
     dmlclib
     faisslib
+    ${BLAS_LIBRARIES}
     )

   if(OPENMP_FOUND)
