diff --git a/build.sh b/build.sh
index 06f0375..bd70d92 100755
--- a/build.sh
+++ b/build.sh
@@ -99,6 +99,7 @@ if hasArg --allgpuarch; then
 fi
 if hasArg --singlegpu; then
     SINGLEGPUPYTHON="--singlegpu"
+    SINGLEGPU="--singlegpu"
     BUILD_CUML_STD_COMMS=OFF
 fi
 if hasArg --nvtx; then
@@ -139,7 +140,8 @@ if (( ${NUMARGS} == 0 )) || hasArg libcuml || hasArg prims || hasArg bench || ha
         GPU_ARCH="-DGPU_ARCHS=ALL"
         echo "Building for *ALL* supported GPU architectures..."
     fi
-
+    
+    export CUDAHOSTCXX=$CXX
     mkdir -p ${LIBCUML_BUILD_DIR}
     cd ${LIBCUML_BUILD_DIR}
 
@@ -150,7 +152,7 @@ if (( ${NUMARGS} == 0 )) || hasArg libcuml || hasArg prims || hasArg bench || ha
           -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
           -DBUILD_CUML_C_LIBRARY=ON \
           -DBUILD_CUML_STD_COMMS=${BUILD_CUML_STD_COMMS} \
-          -DWITH_UCX=ON \
+          -DWITH_UCX=OFF \
           -DBUILD_CUML_MPI_COMMS=OFF \
           -DNVTX=${NVTX} \
           -DPARALLEL_LEVEL=${PARALLEL_LEVEL} \
diff --git a/python/setup.py b/python/setup.py
index 15ed4a0..eeb7346 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -125,7 +125,8 @@ treelite_path = os.path.join(treelite_path, "include")
 # on libcumlprims
 libs = ['cuda',
         'cuml++',
-        'rmm']
+        'rmm',
+        'cudart']
 
 include_dirs = ['../cpp/src',
                 '../cpp/include',
