diff --git a/cpp/test/CMakeLists.txt b/cpp/test/CMakeLists.txt
index efa550d..099215f 100644
--- a/cpp/test/CMakeLists.txt
+++ b/cpp/test/CMakeLists.txt
@@ -77,6 +77,7 @@ if(BUILD_CUML_TESTS)
         ${CUML_C_TARGET}
         pthread
         ${ZLIB_LIBRARIES}
+        ${BLAS_LIBRARIES}
     )

     # (please keep the filenames in alphabetical order)
@@ -133,6 +134,7 @@ if(BUILD_CUML_MG_TESTS)
         ${CUML_CPP_TARGET}
         pthread
         ${ZLIB_LIBRARIES}
+        ${BLAS_LIBRARIES}
     )

     # (please keep the filenames in alphabetical order)
@@ -166,6 +168,7 @@ if(BUILD_PRIMS_TESTS)
         ${ZLIB_LIBRARIES}
         OpenMP::OpenMP_CXX
         Threads::Threads
+        ${BLAS_LIBRARIES}
     )

     # (please keep the filenames in alphabetical order)
